src/plugins/mission_raw/mission_raw_impl.cpp

```diff
@@ -22,16 +22,37 @@ void MissionRawImpl::init()
         MAVLINK_MSG_ID_MISSION_ACK,
         std::bind(&MissionRawImpl::process_mission_ack, this, _1),
         this);
+
+    _parent->register_mavlink_message_handler(
+        MAVLINK_MSG_ID_MISSION_CURRENT,
+        std::bind(&MissionRawImpl::process_mission_current, this, _1),
+        this);
+
+    _parent->register_mavlink_message_handler(
+        MAVLINK_MSG_ID_MISSION_ITEM_REACHED,
+        std::bind(&MissionRawImpl::process_mission_item_reached, this, _1),
+        this);
 }
 
+void MissionRawImpl::enable() {}
+
+void MissionRawImpl::disable() {}
+
 void MissionRawImpl::deinit()
 {
+    _parent->unregister_timeout_handler(_timeout_cookie);
     _parent->unregister_all_mavlink_message_handlers(this);
 }
 
-void MissionRawImpl::enable() {}
-
-void MissionRawImpl::disable() {}
+void MissionRawImpl::reset_mission_progress()
+{
+    std::lock_guard<std::mutex> lock(_mission_data.mutex);
+    _mission_data.last_current_mavlink_mission_item = -1;
+    _mission_data.last_reached_mavlink_mission_item = -1;
```

Comment:
I don't understand the difference between MISSION_CURRENT ( #42 ) and MISSION_ITEM_REACHED ( #46 ) so not sure if the logic to report progress and current progress ans so on is correct

Reply:
Good point. MISSION_REACHED would give you `n` while MISSION_CURRENT is `n+1`.